### DISCUSSION TITLE
Redefining the directory structure [??]

### DISCUSSION THREAD

**Discussion by aingelc12ell** [timschofield/webERP#189] [Development Discussion & Specification]
Is there any plans to re-structure the directories?

Current directory structure exposes everything from the root folder. This requires backend/server configuration to try and hide those that may compromise the installation. 

In the hope of expanding and customizing the system, other plugins are included, and some may be subject to abuse, misuse, or attacks.

Will it be easier and more secure to move the public pages into a `/www` directory while the `/plugins`, `/vendor`, and other dynamic pages to not be directly accessible on the web?

We are all anticipating the newly announced v5 in the making which may include these changes.

**Comment by timschofield** [comment]
Hi Aingel, I understand what you are saying, and in principle I agree. The reason why I haven't is how this transition would work, for existing users, many of whom have have made substantial customisations to their webERP systems over the years.

I would be interested in your thoughts.

Tim

**Comment by aingelc12ell** [comment]
You raise a very valid point: customization. I too see the same by making sure the modules work and will not break anything.

My vision for this is bring to date also the plugins and 3rd party libraries to work.

I will update my fork with the changes, update codebase for php8.3, test functionalities, and see if it can be pushed as another branch.

OR: Anyone from the other devs who may have gone far through the modifications?

**Comment by timschofield** [comment]
The customisations I am talking about are where people have introduced new scripts to their systems that aren't in the main line. We need to consider these things carefully.
What I would suggest is that you make changes incrementally, rather than just create a new branch. You talk about updating the code base for PHP8.3 - I absolutely do want these changes, issue a pull request each time you do an update and I would gratefully accept. Let's work together to make webERP better.

Tim

**Comment by aingelc12ell** [comment]
Very well, I will start with the migration to PHP8.3

Thanks Tim

**Comment by timschofield** [comment]
Great, am looking forward to receiving your pull requests.

Thanks
Tim

On Wed, 23 Oct 2024 at 02:14, Aingel ***@***.***> wrote:

> Very well, I will start with the migration to PHP8.3
>
> Thanks Tim
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/timschofield/webERP/discussions/189#discussioncomment-11023585>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAL6LCQ3AXJ3CE7HXW2E3TZ44BANAVCNFSM6AAAAABQLMCNDCVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMBSGM2TQNI>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***
> com>
>


-- 
www.weberpafrica.com
@TimSchofield2
Blog: https://kwamoja.home.blog/

**Reply by timschofield** [reply]
Sorry, I replied before noticing that you had already submitted a pull request.

Note to self, remember to fully wake up before sending messages :-)

**Reply by aingelc12ell** [reply]
Not an issue with me Tim. All is good, we'll keep moving forward. Thanks!

**Comment by gggeek** [comment]
Since GH does not pop up any links into this discussion: I recently set the baseline (min. supported version) to php 8.1 - see discussion #442

Also, opened a similar issue about restructuring the folder structure in issue #438
